docs/guide/interactions/slash_commands.rst

```diff
@@ -0,0 +1,1232 @@
+
+.. image:: /images/guide/app_commands/repeat_command_described.png
+
+In addition to the decorator, parameter descriptions can also be added using
+Google, Sphinx or Numpy style docstrings.
+
+Examples using a command to add 2 numbers together:
+
+.. code-block:: python
+
+    @client.tree.command() # numpy
+    async def addition(interaction: discord.Interaction, a: int, b: int):
+        """adds 2 numbers together.
+
+        Parameters
+        -----------
+        a: int
+            left operand
+        b: int
+            right operand
+        """
+
+        await interaction.response.send_message(f"{a} + {b} is {a + b}!")
+
+    @client.tree.command() # google
+    async def addition(interaction: discord.Interaction, a: int, b: int):
+        """adds 2 numbers together.
+
+        Args:
+            a (int): left operand
+            b (int): right operand
+        """
+
+    @client.tree.command() # sphinx
+    async def addition(interaction: discord.Interaction, a: int, b: int):
+        """adds 2 numbers together.
+
+        :param a: left operand
+        :param b: right operand
+        """
+
+If both are used, :func:`.app_commands.describe` always takes precedence.
+
+Naming
+^^^^^^^
+
+Since parameter names are confined to the rules of Python's syntax,
+the library offers a method to rename them with the :func:`.app_commands.rename` decorator.
+
+In use:
+
+.. code-block:: python
+    :emphasize-lines: 2
+
+    @client.tree.command()
+    @app_commands.rename(n_times="number-of-times")
+    async def repeat(interaction: discord.Interaction, content: str, n_times: int):
+        to_send = textwrap.shorten(f"{content} " * n_times, width=2000)
+        await interaction.response.send_message(to_send)
+
+When referring to a renamed parameter in other decorators, the original parameter name should be used.
+For example, to use :func:`~.app_commands.describe` and :func:`~.app_commands.rename` together:
+
+.. code-block:: python
+
+    @client.tree.command()
+    @app_commands.describe(
+        content="the text to repeat",
+        n_times="the number of times to repeat the text"
+    )
+    @app_commands.rename(n_times="number-of-times")
+    async def repeat(interaction: discord.Interaction, content: str, n_times: int):
+        to_send = textwrap.shorten(f"{content} " * n_times, width=2000)
+        await interaction.response.send_message(to_send)
+
+Choices
+++++++++
+
+To provide the user with a list of options to choose from for an argument, the :func:`.app_commands.choices` decorator can be applied.
+
+A user is restricted to selecting a choice and can't type something else.
+
+Each individual choice contains 2 fields:
+
+- A name, which is what the user sees
+- A value, which is hidden to the user and only visible to the API. Typically, this is either the same as the name or something more developer-friendly. Value types are limited to either a :class:`str`, :class:`int` or :class:`float`.
```

Comment:
Not 100% sure about my own suggested wording here, but I think it is important to make clear that it is the actual value that gets given to the command by discord.
```suggestion
- A value, which will be the real value provided to the command callback by Discord. Typically, this is either the same as the name or something more developer-friendly. Value types are limited to either a :class:`str`, :class:`int` or :class:`float`.
```

Reply:
Yeah, shows how little I actually work with choices vs. auto-completion. 
I incorrectly assumed it was analogous to auto-completion, which gives you only the actual value of the choice. 

I still find the phrasing of "only visible to the API" sub-optimal, since it reads it to me like it is only visible to the discord api, not your bot, at least in my reading of me putting myself in the position of someone new to the concepts of discord.py or slash commands.
Maybe rephrase to "only visible to your bot" or "only visible to your command callback". Even though I am also iffy on that phrasing entirely. 
just throwing out one last idea here:
```
A value, that should represent the actual value we want to use inside our command callback. 
This can either be the same as the name or something "machine readable"
- like an integer value (see example below)
- a column id in the database
- a mapping key for an internal dictionary
- or any other representation we need.
```